### DISCUSSION TITLE
occassional bluetooth disconnects/rebonds on Huaweii P30

### DISCUSSION THREAD

**Discussion by kameamea** [NightscoutFoundation/xDrip#2017] [Q&A]
xdrip version: alpha
Huaweii P30 with EMUI 11/Android 10
Dexcom G6 with 8M transmitter
Ticwatch E2 (not as collector)

I'm using the recommended settings, but also experimented with the recommendations from similar threads (disable OB1 unbonding).
There is also no energy saving option active for xdrip.

In a good week I may have 2 bluetooth interruptions, on a bad day 5 or more.

When OB1 unbonding is permitted, Android asks for confirmation, but if I miss that, it's even worse, since only a "big reset" helps then.
Thus I've disabled the unbonding as it doesn't help me at all.

Instead from time to time I get from Android: "xdrip wants to disable bluetooth: Confirm". If I confirm I'm asked a bit later to enable it again. But still xdrip doesn't get a connection then.

So the only thing which helps (but also helps reliabliy) is to change the transmitter ID (I just delete the last number) and then enter the same ID again. Then a few minutes later connection works again. I call this "big reset". Whatever xdrip is doing when we change the transmitter ID, this seems to work best for me to reconnect a broken bluetooth connection. And it usually doesn't even bother me with android confirmations about bonding or BT disabling. So I suggest to offer this "big reset" as a button on the System status page (the restart buttons there hardly ever help with my disconnect problems). Or even better we use this instead of or additionally to the OB1 unbonding/bluetooth watchdog automatically when connection doesn't work?

This feature could be helpful, even if we were able to eliminate the current bluetooth connection problems since most likely Android will not stop to make changes which cause trouble to xdrip

**Comment by Navid200** [comment]
Please try the latest Nightly if you are still having connectivity problems:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.06.09
